androidClient/app/src/main/java/com/macbitsgoa/comrades/GetGoogleSignInActivity.java

```diff
@@ -0,0 +1,213 @@
+package com.macbitsgoa.comrades;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.StrictMode;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.Scope;
+import com.google.android.gms.tasks.Task;
+import com.google.api.services.drive.DriveScopes;
+import com.squareup.okhttp.FormEncodingBuilder;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+
+import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
+import static androidx.core.content.PermissionChecker.PERMISSION_DENIED;
+import static androidx.core.content.PermissionChecker.PERMISSION_GRANTED;
+import static com.macbitsgoa.comrades.CHC.TAG_PREFIX;
+
+
+/**
+ * This Activity can be used to make signIn available and ask for storage permissions
+ * at specific places in app.
+ * if you want this activity to return an accessToken,
+ * start activity using ***startActivityForResult(intent,request code)***
+ * By default this Activity will not return anything.
+ * use onActivityResult() in the calling activity to get the result.
+ * use Intent.getExtra() using key as KEY_ACCOUNT to get the user account
+ * and KEY_TOKEN to get access token
+ *
+ * @author Aayush Singla
+ */
+
+public class GetGoogleSignInActivity extends Activity {
+    private static final String TAG = TAG_PREFIX + GetGoogleSignInActivity.class.getSimpleName();
+    private static final int RC_SIGN_IN = 0;
+    public static final String KEY_TOKEN = "token";
+    private boolean returnResult;
+    private static final int ERROR_CODE_PERMISSION_DENIED = 12501;
+    private static final int RC_PERM_REQ_EXT_STORAGE = 7;
```

Comment:
save files on app's internal storage https://developer.android.com/training/data-storage/files#WriteInternalStorage . This will avoid the problem if user (or any nasty "cleaner" app) deleting files accessible to all apps.

Reply:
and no read / write permissions required to do the magic

Reply:
read permission needs to be asked anyhow for uploading files. Downloading file to the app's internal directory is a nice improvement suggestion.Please make an issue for this.

Reply:
Can't it be done now itself?